pkg/apis/events/v1beta1/cloudstoragesource_conversion.go

```diff
@@ -18,17 +18,47 @@ package v1beta1
 
 import (
 	"context"
-	"fmt"
 
+	"github.com/google/knative-gcp/pkg/apis/convert"
+	v1 "github.com/google/knative-gcp/pkg/apis/events/v1"
 	"knative.dev/pkg/apis"
 )
 
 // ConvertTo implements apis.Convertible.
-func (*CloudStorageSource) ConvertTo(_ context.Context, to apis.Convertible) error {
-	return fmt.Errorf("v1beta1 is the highest known version, got: %T", to)
+// Converts source (from v1beta1.CloudStorageSource) into a higher version.
+func (source *CloudStorageSource) ConvertTo(ctx context.Context, to apis.Convertible) error {
+	switch sink := to.(type) {
+	case *v1.CloudStorageSource:
```

Comment:
Just to be sure, you've run this in a cluster and seen that you can read and write v1alpha1, v1beta1, and v1 versions of the resource? 

My guess based on the hub version would have been that everything had to be shoved into the v1alpha1 conversion code. But I've never actually tested that guess. Is the `apis.ConvertToViaProxy` call to handle this case?

Reply:
`apis.ConvertToViaProxy` is to do the conversion between v1 and v1alpha1. I added UT to verify that the conversion between v1, v1beta1 and v1alpha1 works. 
I have not run this in a cluster since this PR only include conversion and relevant helpers, it does not include the webhook and yaml change.
When you say user can read and write v1alpha1, v1beta1, and v1 versions of the resource, do you mean user can run `kubecetl get xxx`?

Reply:
@danyinggu I synced with Adam offline. Currently for this PR it's fine since we have UT, we can verify whether it works with webhook and yaml in your follow up PRs.

Reply:
Sounds good. Thanks for confirming!